src/marine/ndbc_hfradar2ioda.py

```diff
@@ -62,63 +65,73 @@ def _read(self):
-                        self.data[0][locKey][qcKey] = 0
+                        self.data[locKey][valKey] = vals_u[i]
+                        self.data[locKey][errKey] = 0.1
+                        if dopx[i] <= 0.5:
+                            self.data[locKey][qcKey] = 0
+                        else:
+                            self.data[locKey][qcKey] = 11
                     else:
-                        self.data[0][locKey][valKey] = vals_v[i]
-                        self.data[0][locKey][errKey] = 0.1
-                        self.data[0][locKey][qcKey] = 0
+                        self.data[locKey][valKey] = vals_v[i]
+                        self.data[locKey][errKey] = 0.1
+                        if dopy[i] <= 0.5:
+                            self.data[locKey][qcKey] = 0
+                        else:
+                            self.data[locKey][qcKey] = 11
 
 
 def main():
 
-    # Get command line arguments
     parser = argparse.ArgumentParser(
         description=(
-            'Reads marine HF radar radial velocity from archiving'
-            'and converts into IODA formatted'
-            'output files')
+            'HF radar converter to v2'
+        )
     )
 
     required = parser.add_argument_group(title='required arguments')
     required.add_argument(
         '-i', '--input',
-        help="Radar observation input file(s)",
-        type=str, nargs='+', required=True)
+        help="path of radar input file",
+        type=str, required=True)
     required.add_argument(
         '-o', '--output',
-        help="path of ioda output file",
+        help="path of ioda v2 output",
         type=str, required=True)
     required.add_argument(
         '-d', '--date',
         help="base date for the center of the window",
         metavar="YYYYMMDDHH", type=str, required=True)
-
     args = parser.parse_args()
     fdate = datetime.strptime(args.date, '%Y%m%d%H')
-    writer = iconv.NcWriter(args.output, locationKeyList)
-
-    # Read in the altimeter
-    altim = Observation(args.input, fdate, writer)
-
-    # write them out
-    AttrData['date_time_string'] = fdate.strftime("%Y-%m-%dT%H:%M:%SZ")
-
-    (ObsVars, LocMdata, VarMdata) = writer.ExtractObsData(altim.data)
-    writer.BuildNetcdf(ObsVars, LocMdata, VarMdata, AttrData)
+    VarDims = {
+        'sea_water_meridional_current': ['nlocs'],
+        'sea_water_zonal_current': ['nlocs']}
+    radar = Observation(args.input, fdate)
+    GlobalAttrs['date_time_string'] = fdate.strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    ObsVars, nlocs = iconv.ExtractObsData(radar.data, locKeyList)
+    DimDict = {'nlocs': nlocs}
+    writer = iconv.IodaWriter(args.output, locKeyList, DimDict)
+    VarAttrs = DefaultOrderedDict(lambda: DefaultOrderedDict(dict))
+    VarAttrs[('sea_water_meridional_current', 'ObsValue')]['units'] = 'm/s'
+    VarAttrs[('sea_water_zonal_current', 'ObsValue')]['units'] = 'm/s'
+    VarAttrs[('longitude', 'MetaData')]['units'] = 'degree'
```

Comment:
Is there a non default fill value for these data? Also I know the simple example had `VarAttrs` in this location but I think it makes more sense to define units, etc. around line 75 or so where the names are defined.

Reply:
Added fill value for meridional and zonal current to replace default values.
Had a look at @kbhargava 's PR https://github.com/JCSDA-internal/ioda-converters/pull/698
I see it looks better to have VarAttrs following varname in Viirs_13, but this script was written in a different way such that VarAttrs were actually following varname definition that was called by Line 122 radar = Observation(args.input, fdate) , which called class Observation(object). Whereas #698 actually defined VarAttrs within the class, therefore the call was Chl.VarAttrs. I honestly didn't think this is fundamentally different, but let me know if you insist I shall change, then shall I . Thanks.